ios/Classes/TiNfcVTagTechnology.m

```diff
@@ -9,4 +9,701 @@
+}
+
+- (void)fastReadMultipleBlocksWithRequestFlag:(id)args
+{
+  NSNumber *requestFlagsValue = [[args firstObject] valueForKey:@"requestFlags"];
+  uint8_t requestFlags = [requestFlagsValue unsignedCharValue];
+
+  [[self.tagProxy asNFCISO15693Tag] fastReadMultipleBlocksWithRequestFlag:requestFlags
+                                                               blockRange:NSMakeRange(0, 8)
+                                                        completionHandler:^(NSArray<NSData *> *_Nonnull dataBlocks, NSError *_Nullable error) {
+                                                          if (![self _hasListeners:@"didFastReadMultipleBlocksWithRequestFlag"]) {
+                                                            return;
+                                                          }
+                                                          [self fireEvent:@"didFastReadMultipleBlocksWithRequestFlag"
+                                                               withObject:@{
+                                                                 @"errorCode" : error != nil ? NUMINTEGER([error code]) : [NSNull null],
+                                                                 @"errorDescription" : error != nil ? [error localizedDescription] : [NSNull null],
+                                                                 @"errorDomain" : error != nil ? [error domain] : [NSNull null]
+                                                               }];
+                                                        }];
+}
+
+- (void)getSystemInfoAndUIDWithRequestFlag:(id)args
+{
+  NSNumber *requestFlagsValue = [[args firstObject] valueForKey:@"requestFlags"];
+  uint8_t requestFlags = [requestFlagsValue unsignedCharValue];
+
+  [[self.tagProxy asNFCISO15693Tag] getSystemInfoAndUIDWithRequestFlag:requestFlags
+                                                     completionHandler:^(NSData *_Nullable uid, NSInteger dsfid, NSInteger afi, NSInteger blockSize, NSInteger blockCount, NSInteger icReference, NSError *_Nullable error) {
+                                                       if (![self _hasListeners:@"didGetSystemInfoAndUIDWithRequestFlag"]) {
+                                                         return;
+                                                       }
+                                                       [self fireEvent:@"didGetSystemInfoAndUIDWithRequestFlag"
+                                                            withObject:@{
+                                                              @"errorCode" : error != nil ? NUMINTEGER([error code]) : [NSNull null],
+                                                              @"errorDescription" : error != nil ? [error localizedDescription] : [NSNull null],
+                                                              @"errorDomain" : error != nil ? [error domain] : [NSNull null]
+                                                            }];
```

Comment:
You should return all information with payload. Please add **dsfid, afi, blockSize, blockCount and icReference**.

Reply:
Done